cbor/types.ts

```diff
@@ -0,0 +1,70 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+import type { CborArrayDecodedStream } from "./array_decoded_stream.ts";
+import type { CborArrayEncoderStream } from "./array_encoder_stream.ts";
+import type { CborByteDecodedStream } from "./byte_decoded_stream.ts";
+import type { CborByteEncoderStream } from "./byte_encoder_stream.ts";
+import type { CborMapDecodedStream } from "./map_decoded_stream.ts";
+import type { CborMapEncoderStream } from "./map_encoder_stream.ts";
+import type { CborTag } from "./tag.ts";
+import type { CborTextDecodedStream } from "./text_decoded_stream.ts";
+import type { CborTextEncoderStream } from "./text_encoder_stream.ts";
+
+/**
+ * This type specifies the primitive types that the implementation can
+ * encode/decode into/from.
+ */
+export type CborPrimitiveType =
+  | undefined
+  | null
+  | boolean
+  | number
+  | bigint
+  | string
+  | Uint8Array
+  | Date;
+
+/**
+ * This type specifies the encodable and decodable values for
+ * {@link encodeCbor}, {@link decodeCbor}, {@link encodeCborSequence}, and
+ * {@link decodeCborSequence}.
+ */
+export type CborType = CborPrimitiveType | CborTag<CborType> | CborType[] | {
+  [k: string]: CborType;
+};
+
+/**
+ * Specifies the encodable value types for the {@link CborSequenceEncoderStream}
+ * and {@link CborArrayEncoderStream}.
+ */
+export type CborInputStream =
```

Comment:
How about renaming this to `CborStreamInput` as this itself is not always a stream, but an input for the stream?

The same suggestion could also apply to:
- `CborMapInputStream` -> `CborMapStreamInput`
- `CborOutputStream` -> `CborStreamOutput`
- `CborMapOutputStream` -> `CborMapStreamOutput`

Reply:
I'll also work on this change